Framework/Core/include/Framework/DataRefUtils.h

```diff
@@ -168,7 +168,12 @@ struct DataRefUtils {
     if (!header) {
       return 0;
     }
-    return header->payloadSize;
+    // in case of an O2 message with multiple payloads, the size of the message stored
+    // in DataRef is returned
+    if (header->splitPayloadParts > 1 && header->splitPayloadIndex == header->splitPayloadParts) {
```

Comment:
I am not sure I understand this if. Why would `header->splitPayloadIndex == header->splitPayloadParts` matter?

Reply:
we have agreed on with @ironMann that this will be the indication that `splitPayloadParts` payloads will follow. This is a wip solution, think we will later introduce a dedicated header version without member layout change. I will add a comment.